vividus-plugin-rest-api/src/main/java/org/vividus/http/ExtendedHttpLoggingInterceptor.java

```diff
@@ -0,0 +1,142 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import static org.apache.hc.core5.http.ContentType.APPLICATION_JSON;
+import static org.apache.hc.core5.http.ContentType.APPLICATION_XML;
+import static org.apache.hc.core5.http.ContentType.TEXT_HTML;
+import static org.apache.hc.core5.http.ContentType.TEXT_PLAIN;
+import static org.apache.hc.core5.http.ContentType.TEXT_XML;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.EntityDetails;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HeaderElement;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.HttpEntityContainer;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.HttpRequest;
+import org.apache.hc.core5.http.HttpRequestInterceptor;
+import org.apache.hc.core5.http.ProtocolVersion;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.message.MessageSupport;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.spi.LoggingEventBuilder;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.handler.HttpResponseHandler;
+
+public class ExtendedHttpLoggingInterceptor implements HttpRequestInterceptor, HttpResponseHandler
+{
+    private static final String NEW_LINE = "\n";
+    private static final String HEADERS_FORMAT = "\nHeaders:\n{}";
+    private static final String BODY_FORMAT = "\nBody: {}";
+
+    private static final Set<String> LOGGED_CONTENT_TYPES = Set.of(APPLICATION_JSON.getMimeType(),
+            APPLICATION_XML.getMimeType(), TEXT_HTML.getMimeType(), TEXT_PLAIN.getMimeType(), TEXT_XML.getMimeType());
+
+    private final Logger logger;
+
+    private final boolean extendedLogging;
+
+    public ExtendedHttpLoggingInterceptor(Class<?> loggerType, boolean extendedLogging)
+    {
+        this.logger = LoggerFactory.getLogger(loggerType);
+        this.extendedLogging = extendedLogging;
+    }
+
+    @Override
+    public void process(HttpRequest request, EntityDetails entityDetails, HttpContext context) throws IOException
+    {
+        String argBrackets = " {}";
+        LoggingEventBuilder loggingEventBuilder = logger.atInfo();
+        StringBuilder loggerFormat = new StringBuilder("Request:");
+
+        ProtocolVersion protocolVersion = request.getVersion();
+        if (protocolVersion != null)
```

Comment:
is it possible to have no version?

Reply:
> is it possible to have no version?

It seems like it usually null
![image](https://github.com/vividus-framework/vividus/assets/58571912/4322fc1d-2d2d-4e56-acfa-dcd63381eb7a)

Reply:
probably the request is intercepted too early

Reply:
> probably the request is intercepted too early

In debug the protocol version is set only for response, but not for request
Looks like the protocol version needs to be set manually
During req execution HTTP client using HTTP 1.1 as default protocol for request and context
https://github.com/apache/httpcomponents-client/blob/master/httpclient5/src/main/java/org/apache/hc/client5/http/protocol/RequestExpectContinue.java#L69
https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/http/protocol/BasicHttpContext.java#L95